src/commands/list.cr

```diff
@@ -4,15 +4,18 @@ require "./command"
 module Shards
   module Commands
     class List < Command
+      def self.run(path, @@tree = false)
```

Comment:
It doesn't matter much how in the current design, but setting a class variable is actually not very clean. You could just make it a local argument here and it will be forwarded to `super` and `#run`. You can just replace the splat argument `*args` with `tree`.

Reply:
@straight-shoota I don't know. I'm just following @ysbaddaden's request... Do you both agree on this? I'm just getting conflicting inputs from both of you. :confused:

Reply:
@straight-shoota lets avoid asking for stylistic changes, especially when they conflict with previous comments, thanks.

The initial design issue is fixed: keep cli arguments parsing out of command objects.

Reply:
@ysbaddaden I would do that but I don't consider this to be a stylistic issue. It's about putting data in the right place. Conceptually it doesn't make any sense as a class variable.
Also looking at consistency, no other command stores its arguments as class variables.

Reply:
Ah yes, sorry, I shouldn't review on a phone, I was under the misconception that methods were class methods, when we do initialize an instance, so yeah, it should be an instance variable:
```crystal
def self.run(path, tree)
  super
end

def run(path, @tree = false)
end
```

Reply:
Or better: change `Commands.run` method to pass `*args` and `**args`, so named arguments will be forwarded:
```crystal
class Shards::Command
  def self.run(path, *args, **args)
    new(path).run(*args, **args)
  end
end

class Shards::Commands::List
  def run(@tree = false)
  end
end
```

Reply:
`tree` still doesn't even need to be stored as an instance variable either. It's only used inside the method (or rather in `list`).